autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/AmbariAgentHealthEvaluator.java

```diff
@@ -97,12 +107,18 @@ public void execute() {
                         LOGGER.info("Alert: {} is in '{}' state for host '{}'.", AMBARI_AGENT_HEARTBEAT, currentState, hostName);
                     }
                 }
-                if (!hostNamesToRecover.isEmpty()) {
-                    hostNamesToRecover.forEach(hn -> LOGGER.info("Host to recover: {}", hn));
-                    CloudbreakClient cbClient = cloudbreakClientConfiguration.cloudbreakClient();
-                    FailureReport failureReport = new FailureReport();
-                    failureReport.setFailedNodes(hostNamesToRecover);
-                    cbClient.autoscaleEndpoint().failureReport(cluster.getStackId(), failureReport);
+                List<FailedNode> failedNodes = failedNodeRepository.findByClusterId(clusterId);
+                Optional<ChangedNodesReport> changedNodesRequest = getChangedNodes(Set.copyOf(hostNamesToRecover), failedNodes);
+                if (changedNodesRequest.isPresent()) {
+                    LOGGER.info("Nodes state changed for cluster {}. New failed nodes {}, new healthy nodes {}.",
+                            clusterId,
+                            changedNodesRequest.get().getNewFailedNodes(),
+                            changedNodesRequest.get().getNewHealthyNodes());
+                    cloudbreakClientConfiguration.cloudbreakClient()
```

Comment:
Make sure if periscope can't report the failed nodes to cloudbreak, db won't change it's state.

Reply:
Okay, thanks :)